### PR TITLE
Migrate the requestGeoLocation data getter to React Query

### PR DESCRIPTION
**Pull request by jsnajdr**
Migrates the `requestGeoLocation` data getter to React Query.

It's used only once, in the `/me/account` form, when sending analytics event after changing your language. To test, verify that the `calypso_user_language_switch` tracks event is send with a `country_code` property which is your geolocation.

This also fixes a React warning:
```
Warning: Cannot update a component (Connect(MomentProvider)) while rendering a different component
```
that used to be there when loading the `/me/account` page. And was caused by the `requestGeoLocation` selector not only reading from `state`, but also dispatching a Redux action as a side effect. `mapStateToProps` (ran during render) is not a good place for such dispatches.

### CODE REVIEW

**Review by noahtallen — APPROVED** [review]
This tests well for me. I can confirm the `/geo` request happens during pageload.

File: client/data/geo/use-geolocation-query.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { useQuery } from 'react-query';
+
+export const useGeoLocationQuery = () =>
+	useQuery( 'geo', () =>
+		globalThis
+			.fetch( 'https://public-api.wordpress.com/geo/' )
```
Comment (tyxla) [inline client/data/geo/use-geolocation-query.js:6]: It's interesting that we actually need the trailing slash for this endpoint.

File: client/state/data-getters/request-geo-location.js [hunk 2]
```diff
@@ -1,14 +0,0 @@
-import { requestHttpData } from 'calypso/state/data-layer/http-data';
-import { http as rawHttp } from 'calypso/state/http/actions';
-
-const requestGeoLocation = () =>
```
Comment (tyxla) [inline client/state/data-getters/request-geo-location.js:4]: This is still included in the `httpData` example: https://github.com/Automattic/wp-calypso/blob/migrate/request-geolocation/client/state/data-layer/http-data.ts#L249, should we replace it with another one before 🚢 ?

**Review by tyxla — APPROVED** [review]
LGTM 👍 

Just one tiny bit to address before 🚢

File: client/data/geo/use-geolocation-query.js (hunk 1 above)
Comment (jsnajdr) [reply client/data/geo/use-geolocation-query.js:6]: Yeah I tried to remove it and failed miserably :)

File: client/state/data-getters/request-geo-location.js (hunk 2 above)
Comment (jsnajdr) [reply client/state/data-getters/request-geo-location.js:4]: We're going to remove `http-data`, hopefully soon, and no new usages should be being added. That's why I didn't want to spend time inventing another example.

**Pull request merged**